api/v3/Mailing.php

```diff
@@ -613,20 +614,34 @@ function civicrm_api3_mailing_send_test($params) {
   if (!array_key_exists('test_group', $params) && !array_key_exists('test_email', $params)) {
     throw new API_Exception("Mandatory key(s) missing from params array: test_group and/or test_email field are required");
   }
-  civicrm_api3_verify_mandatory($params,
-    'CRM_Mailing_DAO_MailingJob',
-    array('mailing_id'),
-    FALSE
-  );
+  $testEmailParams = [
+    'mailing_id' => $params['mailing_id'],
+    'is_test' => 1,
+    'status' => 'Scheduled',
+    'scheduled_date' => CRM_Utils_Date::processDate(date('Y-m-d'), date('H:i:s')),
+  ];
 
-  $testEmailParams = _civicrm_api3_generic_replace_base_params($params);
-  $testEmailParams['is_test'] = 1;
-  $testEmailParams['status'] = 'Scheduled';
-  $testEmailParams['scheduled_date'] = CRM_Utils_Date::processDate(date('Y-m-d'), date('H:i:s'));
   $job = civicrm_api3('MailingJob', 'create', $testEmailParams);
   $testEmailParams['job_id'] = $job['id'];
-  $testEmailParams['emails'] = array_key_exists('test_email', $testEmailParams) ? explode(',', $testEmailParams['test_email']) : NULL;
+  $testEmailParams['emails'] = array();
+
+  $emailDetail = array();
```

Comment:
@jmcclelland jamie i think the original handling of `$testEmailParams['emails'] ` is more correct and i think rather than all this code all that needs to happen is that `test_group` param is merged into the `$testEmailParams`

Reply:
Ahhh. That would make a lot more sense. I couldn't figure out how this ever worked.